src/main/java/frc/robot/arm/Arm.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.arm;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Arm extends SubsystemBase {
+    private double targetAngle;
+    private PIDController pid;
+    private boolean broken;
+    private ShuffleboardTab dashboard;
+    private IArmIO armIO;
+
+    public Arm(IArmIO armIO) {
+        this.armIO = armIO;
+        pid = new PIDController(Constants.Arm.PID_CONTROLLER_P_COEFFICIENT, Constants.Arm.PID_CONTROLLER_I_COEFFICIENT, Constants.Arm.PID_CONTROLLER_D_COEFFICIENT);
+        broken = false;
+        initDashboard();
+    }
+
+    public double getAngle() {
+        return armIO.getMeasurement();
+    }
+
+    public boolean isOnTarget() {
+        return ((targetAngle - Constants.Arm.TARGET_MOVEMENT_RANGE_DEGREES <= getAngle()) && (targetAngle + Constants.Arm.TARGET_MOVEMENT_RANGE_DEGREES >= getAngle()));
+    }
+
+    public double getTarget() {
+        return targetAngle;
+    }
+
+    public void setTarget(double inputTarget) {
+        if (inputTarget >= Constants.Arm.MAX_ARM_RANGE_DEGREES) {
+            targetAngle = Constants.Arm.MAX_ARM_RANGE_DEGREES;
+        } else {
+            targetAngle = inputTarget;
+        }
+    }
+
+    public boolean isAtHomePostion() {
+        return armIO.atFrontLimit();
+    }
+
+    public boolean isBroken() {
+        return broken;
+    }
+
+    private String armPositionName() {
+        if (targetAngle == Constants.Arm.HIGH_SCORE_POSITION){
+            return ("High Goal");
+        } else if (targetAngle == Constants.Arm.LOW_SCORE_POSITIOIN){
+            return ("Low Goal");
+        } else if (targetAngle == Constants.Arm.HOME_POSITION) {
+            return ("Home");
+        } else{
+            return ("Moving...");
```

Comment:
You should never get here. This method is checking the target, not the current angle. Based on your current commands the return of this method should always be one of your first three conditions. 

What you should change depends on what you want the method to actually do. If you want it to tell the target, you can just make the return of the else something like `"Impossible"` or `"Undefined target"`. If you want the method to tell you where you really are, switch what angle you are looking at.